components/org.wso2.carbon.integrator.core/src/main/java/org/wso2/carbon/integrator/core/handler/RESTBasicAuthHandler.java

```diff
@@ -0,0 +1,110 @@
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.core.axis2.Axis2Sender;
+import org.apache.synapse.rest.Handler;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.Map;
+
+public class RESTBasicAuthHandler implements Handler {
+
+    private static final Log log = LogFactory.getLog(RESTBasicAuthHandler.class);
+
+    @Override
+    public boolean handleRequest(MessageContext messageContext) {
+
+        org.apache.axis2.context.MessageContext axis2MessageContext
+                = ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        Object headers = axis2MessageContext.getProperty(
+                org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+
+        if (headers != null && headers instanceof Map) {
+            Map headersMap = (Map) headers;
+            if (headersMap.get(HTTPConstants.HEADER_AUTHORIZATION) == null) {
+                headersMap.clear();
+                axis2MessageContext.setProperty(BasicAuthConstants.HTTP_STATUS_CODE, BasicAuthConstants.SC_UNAUTHORIZED);
+                headersMap.put(BasicAuthConstants.WWW_AUTHENTICATE, BasicAuthConstants.WWW_AUTH_METHOD);
+                axis2MessageContext.setProperty(BasicAuthConstants.NO_ENTITY_BODY, true);
+                messageContext.setProperty(BasicAuthConstants.RESPONSE, BasicAuthConstants.TRUE);
+                messageContext.setTo(null);
+                Axis2Sender.sendBack(messageContext);
+                return false;
+
+            } else {
+                String authHeader = (String) headersMap.get(HTTPConstants.HEADER_AUTHORIZATION);
+                String credentials = authHeader.substring(6).trim();
+                if (processSecurity(credentials)) {
+                    return true;
+                } else {
+                    headersMap.clear();
+                    axis2MessageContext.setProperty(BasicAuthConstants.HTTP_STATUS_CODE, BasicAuthConstants.SC_FORBIDDEN);
+                    axis2MessageContext.setProperty(BasicAuthConstants.NO_ENTITY_BODY, true);
+                    messageContext.setProperty(BasicAuthConstants.RESPONSE, BasicAuthConstants.TRUE);
+                    messageContext.setTo(null);
+                    Axis2Sender.sendBack(messageContext);
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean handleResponse(MessageContext messageContext) {
+
+        return true;
+    }
+
+    @Override
+    public void addProperty(String s, Object o) {
+
+    }
+
+    @Override
+    public Map getProperties() {
+
+        return null;
+    }
+
+    public boolean processSecurity(String credentials) {
```

Comment:
Looking at the functionality of this method seems to me that "authenticate" or "authenticateUser" is more appropriate as the name of the method.